assets/js/util/index.js

```diff
@@ -303,11 +303,11 @@ export const refreshAuthentication = async () => {
 		} );
 
 		// We should really be using state management. This is terrible.
-		window.googlesitekit.setup = window.googlesitekit.setup || {};
-		window.googlesitekit.setup.isAuthenticated = response.isAuthenticated;
-		window.googlesitekit.setup.requiredScopes = response.requiredScopes;
-		window.googlesitekit.setup.grantedScopes = response.grantedScopes;
-		window.googlesitekit.setup.needReauthenticate = requiredAndGrantedScopes.length < response.requiredScopes.length;
+		global.googlesitekit.setup = global.googlesitekit.setup || {};
+		global.googlesitekit.setup.isAuthenticated = response.isAuthenticated;
+		global.googlesitekit.setup.requiredScopes = response.requiredScopes;
+		global.googlesitekit.setup.grantedScopes = response.grantedScopes;
+		global.googlesitekit.setup.needReauthenticate = requiredAndGrantedScopes.length < response.requiredScopes.length;
```

Comment:
Not related to this issue but this condition seems weak.

Reply:
This is my favorite worst part of the codebase, and the most obvious place for why we need a datastore. When this "fix" was made, there was no time for more though.